dm/pkg/binlog/event/util.go

```diff
@@ -354,6 +354,22 @@ func GetParserForStatusVars(statusVars []byte) (*parser.Parser, error) {
 	return parser2, err
 }
 
+// GetServerCollationByStatusVars gets server collation by binlog statusVars.
+func GetServerCollationByStatusVars(statusVars []byte) (string, error) {
+	vars, err := statusVarsToKV(statusVars)
+	b, ok := vars[QCharsetCode]
+
+	if !ok {
+		if err == nil {
+			// only happen when this is a dummy event generated by DM
+			err = fmt.Errorf("Q_CHARSET_CODE not found in status_vars %v", statusVars)
+		}
+		return mysql.UTF8MB4DefaultCollation, err
```

Comment:
when we can't find the collation, I prefer we return the default value of MySQL not TiDB, which is "utf8mb4_general_ci"

https://docs.pingcap.com/tidb/stable/character-set-and-collation

Reply:
ok